### DISCUSSION TITLE
Anthropic + OpenAI chats are not responding - anyone else having this issue?

### DISCUSSION THREAD

**Discussion by fluxgravity** [AllYourBot/hostedgpt#496] [General]
Hi,

I recently deployed a fork of the app to Heroku. I think I have everything set up correctly including my default API keys. However, my chats are not giving me proper responses. With my Claude assistants, the responses never generate. And with my GPT assistant (3.5 Turbo), I receive a quota error.

Currently, I'm on the Free tiers for both API plans, which allows me limited usage on chat models. I am taking this into account when I set up my assistants and am not using chat models that are not available to me on the Free tiers. I'm okay with upgrading to a paid tier but would like to avoid that cost if my usage does not justify it.

Has anyone run into a similar issue? Any tips?

Thanks!

**Comment by keithschacht** [comment]
At first this sounded like a technical issue, like maybe the background worker isn't running (because that's what streams the response). When you say you get a quota error, what exactly is the error? Can you copy and paste that? That could help me to confirm.

But my guess is that this is not technical and is a billing issue. Both Anthropic and OpenAI charge to use their APIs. I believe they both give you $5 - 20 initial API credits so you can try them for free but there is no ongoing free tier for them. They do have free usage in their web-based interfaces but that doesn't extend to the API.

Based on what you described I'm guessing that the API keys aren't properly set up with billing information. My first suggestion:

- Click your profile picture in bottom left of the app and go to settings
- Click on API services and go to both OpenAI and Anthropic
- In the "Token" field there is a "How do I get?" link. Click that again and review all those steps to make sure you're getting the token from the right place. And while you're in each service, make sure you have a credit card on file and look for any settings about $ limits which might be set or auto-fill limits.

**Comment by fluxgravity** [comment]
That did the trick. With OpenAI, it turned out that my free credits expired a while back - the view for the credits is a bit hidden. After adding my credit card, I started getting responses to my prompts in HostedGPT. For posterity, I've attached the initial rate quota error.

<img width="1036" alt="rate_quota_error" src="https://github.com/user-attachments/assets/de4efcd0-dea3-4c08-875e-382a54554733">

For Anthropic, I had to claim the free $5 credit to take advantage of their free tier. But I'll need to enter billing info eventually as the free credits also expire.

I appreciate your help with troubleshooting. I'm looking forward to using the app as my main AI tool.
